runtime/src/main/java/org/corfudb/runtime/collections/CorfuTable.java

```diff
@@ -78,18 +78,18 @@
 
     // The "main" map which contains the primary key-value mappings.
     private final ContextAwareMap<K, V> mainMap;
-    private final Set<Index.Spec<K, V, ? extends Comparable>> indexSpec;
-    private final Map<String, Map<Comparable, Map<K, V>>> secondaryIndexes;
+    private final Set<Index.Spec<K, V, ?>> indexSpec;
```

Comment:
missing getByIndex and getByIndexAndFilter + unit test

Reply:
Done

Reply:
Done